File: LeetCode/Miscellaneous/divide two integers.py
class Solution:
    def divide(self, dividend: int, divisor: int) -> int:
        try:
            if dividend != 0:
                ret = int(dividend / divisor)
                if ret >= 2**31:
                    return ret - 1
                elif ret < -2 ** 31:
                    return ret + 1
                else:
                    return ret
            else:
                return 0
        except Exception as error:
            print(f"[ERROR] divide: {error}")

File: LeetCode/Bit Manipulation/Divide Two Integers.cpp
class Solution
{
public:
    int divide(int dividend, int divisor)
    {
        if (dividend == INT_MIN && divisor == -1)
            return INT_MAX;
        long a = abs(dividend), b = abs(divisor), res = 0;
        while (a - b >= 0)
        {
            int x = 0;
            while ((a - (b << x << 1)) >= 0)
                x++;
            res += 1 << x;
            a -= b << x;
        }
        return (dividend > 0) == (divisor > 0) ? res : -res;
    }
};